File: src/backend/ResumeMatcher.Domain/Models/AnalysisResultModel.cs
namespace ResumeMatcher.Domain;

public sealed record AnalysisResultModel(
    Guid Id,
    Guid ResumeId,
    double OverallScore,
    double SkillsScore,
    double ExperienceScore,
    double SeniorityScore,
    double RequirementsScore,
    double EducationScore,
    IReadOnlyList<EvidenceItemModel> MatchedSkills,
    IReadOnlyList<EvidenceItemModel> MissingSkills,
    IReadOnlyList<EvidenceItemModel> RequirementsMet,
    IReadOnlyList<EvidenceItemModel> RequirementsMissing,
    IReadOnlyList<EvidenceItemModel> Strengths,
    IReadOnlyList<EvidenceItemModel> PointsOfAttention,
    IReadOnlyList<EvidenceItemModel> Recommendations);

File: src/frontend/src/features/analysis/models/AnalysisResultModel.ts
export interface EvidenceItemModel {
  text: string
  evidence?: string | null
}

export interface AnalysisResultModel {
  id: string
  resumeId: string
  overallScore: number
  skillsScore: number
  experienceScore: number
  seniorityScore: number
  requirementsScore: number
  educationScore: number
  matchedSkills: EvidenceItemModel[]
  missingSkills: EvidenceItemModel[]
  requirementsMet: EvidenceItemModel[]
  requirementsMissing: EvidenceItemModel[]
  strengths: EvidenceItemModel[]
  pointsOfAttention: EvidenceItemModel[]
  recommendations: EvidenceItemModel[]
}
